File: java/com/example/Eco/viewmodel/SignupViewModel.kt
package com.example.Eco.viewmodel

import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel
import com.example.Eco.data.SignupRequest
import com.example.Eco.data.SignupResponse
import com.example.Eco.retrofit.RetrofitServiceInterface
import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response

class SignupViewModel(private val retrofitService: RetrofitServiceInterface) : ViewModel() {

    val signupResponse = MutableLiveData<SignupResponse>()
    val errorMessage = MutableLiveData<String>()

    // Fonction pour effectuer l'inscription
    fun signup(name: String, email: String, password: String) {
        val request = SignupRequest(name, email, password)

        // Appel de l'API d'inscription via Retrofit
        retrofitService.signup(request).enqueue(object : Callback<SignupResponse> {
            override fun onResponse(call: Call<SignupResponse>, response: Response<SignupResponse>) {
                if (response.isSuccessful) {
                    // Inscription réussie, mise à jour de signupResponse
                    signupResponse.value = response.body()
                } else {
                    // Échec de l'inscription, mise à jour du message d'erreur
                    errorMessage.value = "Signup failed: ${response.message()}"
                }
            }

            override fun onFailure(call: Call<SignupResponse>, t: Throwable) {
                // En cas d'erreur de réseau ou autre
                errorMessage.value = t.message
            }
        })
    }
}

File: lib/viewmodel/signup_view_model.dart
import 'package:flutter/material.dart';
import '../models/signup_request.dart';
import '../models/signup_response.dart';
import '../services/api_service.dart';

class SignupViewModel extends ChangeNotifier {
  final ApiService apiService;

  SignupViewModel({required this.apiService});

  SignupResponse? _signupResponse;
  String? _errorMessage;

  SignupResponse? get signupResponse => _signupResponse;
  String? get errorMessage => _errorMessage;

  Future<void> signup(String name, String email, String password) async {
    try {
      final request = SignupRequest(name: name, email: email, password: password);
      _signupResponse = await apiService.signup(request);
      _errorMessage = null;
    } catch (e) {
      _signupResponse = null;
      _errorMessage = e.toString();
    }
    notifyListeners();
  }
}